tests/testthat/test-mergedatasetsbycase.R

```diff
@@ -0,0 +1,794 @@
+context("MergeDataSetsByCase")
+
+library(gtools)
+
+findInstDirFile <- function(file)
+{
+    file.path(system.file("testdata", package = "flipData", mustWork = TRUE),
+              file)
+}
+
+# cola1.sav: Q2 (Q2. Gender) removed, Q3_3 (Q3. Age in years) moved down to be
+#            after Q4_C_2. Variables after Q4_C_4 removed.
+# cola2.sav: Q2 (Q2. Gender) category labels changed to M, F, and
+#            Q3_3 (Q3. Age in years) category values changed to 1-10.
+#            Variables after Q4_C_4 removed.
+# cola3.sav: Q1.  Fragments coded variables removed, variables after Q4_C_2
+#            removed except for Q1_F (Q1.  Fragments - Coke) which is a text variable.
+# cola4.sav: same as cola3.sav except that Q3_3 renamed to Q3_3_new_name,
+# cola5.sav: Q4_A_3, Q4_B_2, Q4_C_2 renamed to Q4_A_3_new, Q4_B_2_new, Q4_C_2_new,
+#            variables after Q4_C_4 removed.
+# cola6.sav: Q2 (Q2. Gender) converted to numeric
+# cola7.sav: Q11_l (text variable) renamed to Q3_3
+# cola8.sav: Q1_F_c renamed to Q1F_c
+# cola9.sav: Only keep Q1_*, Q3_3, Q4_A_3, Q4_B_2, Q4_C_2, with "Q" in variable
+#            names replaced with "Question".
+# cola10.sav: Only keep Q1_*, Q3_3, Q4_A_3, Q4_B_2, Q4_C_2.
+# cola11.sav: Only keep Q1_*, Q3_3, Q4_A_3, Q4_B_2, Q4_C_2, labels for Q1_*
+#             are lower-case, Q3_3 renamed to Q33 and underscores removed from Q4*.
+# cola12.sav: Only keep Q1_*, Q3_3, Q4_A_3, Q4_B_2, Q4_C_2. "Coca Cola"
+#             replaced with "Coke" in Q4_A_3 label. Value labels
+#             "2 - 3 days a week" and "4 - 5 days a week".
+# cola13.sav: Includes Date variable LastResp_date.
+# cola14.sav: Includes text variable LastResp_date_text containing strings that can be parsed
+#             as dates.
+#
+# To update the merge.data.set.output.rda test data in flipFormat,
+# simply save merge.data.set.output from this test
+# into flipFormat/inst/testdata/merge.data.set.output.rda
+test_that("Example used for widget test in flipFormat", {
+    expect_error(merge.data.set.output <- MergeDataSetsByCase(data.set.names = c(findInstDirFile("cola1.sav"),
+                                                                                 findInstDirFile("cola2.sav"),
+                                                                                 findInstDirFile("cola5.sav"),
+                                                                                 findInstDirFile("cola8.sav")),
+                                                              data.sets.whose.variables.are.kept = 1,
+                                                              variables.to.combine = "Q4_A_3,Q4_A_3_new"), NA)
+})
```

Comment:
When NA is supplied as the second argument to expect_error, no error is expected. I know it may seem misleading without this information. I'm not aware of a better way to do this and this is commonly done in flip* unit tests.